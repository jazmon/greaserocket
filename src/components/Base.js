// @flow
import React from 'react';
import { View, StatusBar, StyleSheet } from 'react-native';
import { DARK_PRIMARY_COLOR, LIGHT_STATUS_BAR } from '../constants/theme';

type Props = {
  style: Style,
};

const Base = ({ children, style }: Props): ElementType => (
  <View style={[styles.container, style]}>
    <StatusBar
      backgroundColor={DARK_PRIMARY_COLOR}
      barStyle={LIGHT_STATUS_BAR ? 'light-content' : 'dark-content'}
    />
    {children}
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});

export default Base;
