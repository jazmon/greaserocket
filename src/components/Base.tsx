import React, { ReactChildren } from 'react';
import { StatusBar, ViewStyle } from 'react-native';
import styled, { withTheme } from 'styled-components/native';
import { getStatusBarTextColor, ThemeType } from 'constants/theme';
import { Maybe } from 'types';

interface Props {
  style?: ViewStyle;
  children?: Maybe<ReactChildren>;
  theme: any;
}

const Container = styled.View`
  flex: 1;
  background-color: ${({ theme }: { theme: any }) => theme.colors.white};
`;

const Base = ({ children, style, theme }: Props) =>
  <Container style={style}>
    <StatusBar
      backgroundColor={theme.colors.primary.dark}
      barStyle={getStatusBarTextColor(theme.colors.primary.dark)}
    />
    {children}
  </Container>;

export default withTheme(Base);
