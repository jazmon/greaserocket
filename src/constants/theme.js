// @flow
import color from 'color';

const theme = {
  DARK_PRIMARY_COLOR: '#388E3C',
  DEFAULT_PRIMARY_COLOR: '#4CAF50',
  LIGHT_PRIMARY_COLOR: '#C8E6C9',
  TEXT_PRIMARY_COLOR: '#FFFFFF',
  ACCENT_COLOR: '#FFC107',
  PRIMARY_TEXT_COLOR: 'rgba(0, 0, 0, 0.87)',
  SECONDAY_TEXT_COLOR: 'rgba(0, 0, 0, 0.54)',
  DIVIDER_COLOR: 'rgba(0, 0, 0, 0.12)',
  DISABLED_TEXT_COLOR: 'rgba(0, 0, 0, 0.38)',
  WHITE: '#FFF',
  BLACK: '#000',
};

export default {
  ...theme,
  LIGHT_STATUS_BAR: color(theme.DARK_PRIMARY_COLOR).dark(),
};
