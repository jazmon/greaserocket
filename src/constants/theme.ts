// @flow
import color from 'color';

const theme = {
  colors: {
    primary: {
      normal: '#4CAF50',
      dark: '#388E3C',
      light: '#C8E6C9',
    },
    accent: {
      normal: '#FFC107',
    },
    white: '#FFFFFF',
    black: '#000000',
    transparent: 'transparent',
  },
  text: {
    colors: {
      light: '#FFFFFF',
      primary: 'rgba(0, 0, 0, 0.87)',
      secondary: 'rgba(0, 0, 0, 0.54)',
      disabled: 'rgba(0, 0, 0, 0.38)',
      divider: 'rgba(0, 0, 0, 0.12)',
    },
    size: 12,
  },
};

export type ThemeType = typeof theme;

export function getStatusBarTextColor(
  backgroundColor: string = 'light-content',
): 'light-content' | 'dark-content' {
  return color(backgroundColor).dark() ? 'light-content' : 'dark-content';
}

export default theme;
