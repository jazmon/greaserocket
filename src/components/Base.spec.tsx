import 'react-native';
import * as React from 'react';
// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from 'constants/theme';

import Base from './Base';

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <Base />
    </ThemeProvider>,
  );
  expect(tree).toMatchSnapshot();
});
