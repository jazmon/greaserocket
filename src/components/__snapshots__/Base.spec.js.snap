// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
/>
`;
